### PR TITLE
count total length of files in dir stat

### PR DESCRIPTION
**Pull request by Hexilee**

### CODE REVIEW

File: pkg/meta/redis.go [hunk 1]
```diff
@@ -528,6 +528,10 @@ func (m *redisMeta) usedSpaceKey() string {
 	return m.prefix + usedSpace
 }
 
+func (m *redisMeta) dirFileLengthKey() string {
+	return m.prefix + "dirFileLength"
```
Comment (SandyXSD) [inline pkg/meta/redis.go:532]: "dirDataLength"

File: pkg/meta/base.go [hunk 2]
```diff
@@ -277,50 +278,49 @@ func (m *baseMeta) groupBatch(batch map[Ino]dirStat, size int) [][]Ino {
...
 		err = errors.Wrap(eno, "calc dir stat")
 		return
 	}
 	for _, e := range entries {
 		inodes += 1
-		var newSpace uint64
+		var l uint64
 		if e.Attr.Typ == TypeFile {
-			newSpace = e.Attr.Length
+			l = e.Attr.Length
```
Comment (SandyXSD) [inline pkg/meta/base.go:291]: also add length for symlink

File: pkg/meta/redis.go (hunk 1 above)
Comment (Hexilee) [reply pkg/meta/redis.go]: Renamed

File: pkg/meta/base.go (hunk 2 above)
Comment (Hexilee) [reply pkg/meta/base.go:294]: After this PR, both of the fast summary and strict summary won't count the length of symlink.

File: pkg/meta/base.go [hunk 3]
```diff
@@ -280,57 +281,56 @@ func (m *baseMeta) groupBatch(batch map[Ino]dirStat, size int) [][]Ino {
 	return batches
 }
 
-func (m *baseMeta) calcDirStat(ctx Context, ino Ino) (space, inodes uint64, err error) {
+func (m *baseMeta) calcDirStat(ctx Context, ino Ino) (length, space, inodes uint64, err error) {
```
Comment (davies) [inline pkg/meta/base.go:284]: we should use `dirStat` instead of `length, space, inodes uint64`

Comment (davies) [reply pkg/meta/base.go:284]: Maybe we can do the refactor first, then it will be easy to add `length`

Comment (Hexilee) [reply pkg/meta/base.go]: Done

**Pull request merged**